src/taipy/core/data/excel.py

```diff
@@ -237,56 +234,46 @@ def _read_as_numpy(self):
             return {sheet_name: df.to_numpy() for sheet_name, df in sheets.items()}
         return sheets.to_numpy()
 
-    def _read_as_pandas_dataframe(self, sheet_names=None) -> Union[Dict[Union[int, str], pd.DataFrame], pd.DataFrame]:
+    def _do_read_excel(self, engine, sheet_names, kwargs) -> pd.DataFrame:
+        df = pd.read_excel(
+            self._path,
+            sheet_name=sheet_names,
+            **kwargs,
+        )
+        if isinstance(df, dict):
+            for key, value in df.items():
+                df[key] = pd.DataFrame(value) if engine == "pandas" else modin_pd.DataFrame(value)
```

Comment:
I take that this is to simplify the read excel file with modin process, which was messy since the beginning (the issue I raised with the modin team is still not fixed 😞 ). `read_excel` will return a dict of {'sheet_name': pandas.DataFrame}` if the excel file has multiple sheet names 😃 
```suggestion
                df[key] = value if engine == "pandas" else modin_pd.DataFrame(value)
```